src/Stack/Build/ConstructPlan.hs

```diff
@@ -185,6 +185,10 @@ constructPlan :: forall env. HasEnvConfig env
 constructPlan ls0 baseConfigOpts0 locals extraToBuild0 localDumpPkgs loadPackage0 sourceMap installedMap initialBuildSteps = do
     logDebug "Constructing the build plan"
 
+    bconfig <- view buildConfigL
+    when (hasBaseInDeps bconfig) $
+      prettyWarn $ flow "You are trying to upgrade/downgrade base, which is almost certainly not what you really want. Please, consider using another GHC version if you need a certain version of base, or removing base from extra-deps." <> line
```

Comment:
Great warning, but I think some new users may still be confused. How about including a link to the original issue so they can learn more?